publish.js

```diff
@@ -266,21 +266,37 @@ publish.tagVC = function (isTest, version, options) {
 };
 
 /**
- * Restore the package.json file to the latest committed version.
+ * Restore the files specified in filesToClean option to the latest committed version.
+ * Restores package.json and package-lock.json files to the latest commited version by default.
  *
- * Used internally to reset version number changes in package.json
- * @param moduleRoot {String} - the directory where the package.json file to
+ * Used internally to reset version number changes in list of files in filesToClean option.
+ * First checks whether the list of files in filesToClean option are being tracked by git or not.
+ * @param moduleRoot {String} - the directory where the files to
                                 clean is located in.
- * @param options {Object} - e.g. {"cleanCmd": "git checkout -- package.json"}
+ * @param options {Object} - e.g. {"checkFileTracking": "git ls-files --error-unmatch ${file}", "cleanCmd": "git checkout -- ${file}", "filesToClean": "package.json, package-lock.json"}
  */
 publish.clean = function (moduleRoot, options) {
     var originalDir = process.cwd();
 
     // change to the module root directory
     process.chdir(moduleRoot || "./");
 
-    // run the clean command
-    publish.execSyncFromTemplate(options.cleanCmd);
+    var file = options.filesToClean.split(/\s*,\s*/);
+
+    file.forEach(function (file) {
+
+        // check whether file is being tracked or not and then try to clean it
+        try {
+            publish.execSyncFromTemplate(options.checkFileTracking, {
```

Comment:
I understand what you were trying to do here, but this new check does nothing.  There is no return value from the call and you don't actually use that to decide whether to clean the file.

Reply:
Or perhaps you were counting on an error to be thrown if the file was untracked?  If that's the case, a comment would be a good idea.

Reply:
OK, I had to learn a bit more about how this package works myself (that's normal).  I see that it would have thrown an error if the file wasn't tracked, that part would be more or less OK with a reasonable comment.